apps/prairielearn/src/lib/editors.js

```diff
@@ -82,134 +82,152 @@ export class Editor {
+          await cleanAndResetRepository(this.course, gitEnv, job);
+        }
 
-              if (!config.fileEditorUseGit) {
-                await syncCourseFromDisk(this.course, startGitHash, job);
-                return;
-              }
+        try {
+          job.info('Write changes to disk');
+          await this.write();
+        } catch (err) {
+          if (config.fileEditorUseGit) {
+            await cleanAndResetRepository(this.course, gitEnv, job);
+          }
 
-              // Safety check: make sure the course has a defined branch and repository.
-              if (!this.course.branch || !this.course.repository) {
-                job.fail('Git repository or branch are not set for this course. Exiting...');
-                return;
-              }
+          throw err;
+        }
 
-              try {
-                await job.exec('git', ['add', ...this.pathsToAdd], {
-                  cwd: this.course.path,
-                  env: gitEnv,
-                });
-                await job.exec(
-                  'git',
-                  [
-                    '-c',
-                    `user.name="${this.user.name}"`,
-                    '-c',
-                    `user.email="${this.user.uid}"`,
-                    'commit',
-                    '-m',
-                    this.commitMessage,
-                  ],
-                  {
-                    cwd: this.course.path,
-                    env: gitEnv,
-                  },
-                );
-              } catch (err) {
-                await cleanAndResetRepository(this.course, gitEnv, job);
-                throw err;
-              }
+        // Safety check: make sure the course has a defined branch and repository.
+        if (!this.course.branch || !this.course.repository) {
+          job.fail('Git repository or branch are not set for this course. Exiting...');
+          return;
+        }
 
-              try {
-                job.data.pushAttempted = true;
-
-                await job.exec('git', ['push'], {
-                  cwd: this.course.path,
-                  env: gitEnv,
-                });
-
-                // We'll look for this flag on the `editError` page to know if
-                // we need to display instructions to recover from a failed push.
-                job.data.pushSucceeded = true;
-              } finally {
-                // Regardless of whether we error, we'll do a clean and reset:
-                //
-                // If pushing succeeded, the clean will remove any empty directories
-                // that might have been left behind by operations like renames.
-                //
-                // If pushing errored, the reset will get us back to a known good state.
-                await cleanAndResetRepository(this.course, gitEnv, job);
-              }
+        if (!config.fileEditorUseGit) {
+          await syncCourseFromDisk(this.course, startGitHash, job);
+          return;
+        }
 
```

Comment:
This block should not have been moved; it's important that we only check for `course.branch` and `course.repository` if we're actually going to use Git during syncing.

Reply:
Moved this check for git back to before we check for `course.branch` and `course.repository` in [427ba97](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/427ba9786ae9ed763d782425b39575cacf18bb96).